libnetwork/iptables/iptables.go

```diff
@@ -116,18 +116,20 @@ func detectIptables() {
 		return
```

Comment:
> looks like there's a lot of cleaning up to do in this part of the code

I'd be happy to provide some help. Although, I'm not sure what could be done to improve this package. Do you have some ideas?

Reply:
> This would allow to look for (and use only) ip6tables, eg. to have Docker supports running without ipv4 (I started working on that few weeks ago).

Looks like there's actually a feature request for that (#32850).